tests/test_editable.py

```diff
@@ -156,3 +156,191 @@ def test_install_dir(isolated):
+        pytest.param(
+            True,
+            "redirect",
+            "any(str(x).endswith('.so') for x in files(pkg).iterdir())",
+            marks=pytest.mark.xfail,
+        ),
+        (True, "redirect", "isinstance(files(pkg.subpkg1), pathlib.Path)"),
+        pytest.param(
+            True,
+            "redirect",
+            "any(str(x).endswith('.so') for x in files(pkg.subpkg1).iterdir())",
+            marks=pytest.mark.xfail,
+        ),
+        (True, "redirect", "isinstance(files(pkg.subpkg2), pathlib.Path)"),
+        pytest.param(
+            True,
+            "redirect",
+            "any(str(x).endswith('.so') for x in files(pkg.subpkg2).iterdir())",
+            marks=pytest.mark.xfail,
+        ),
+        (True, "redirect", "isinstance(files(pkg.subpkg2.subsubpkg1), pathlib.Path)"),
+        pytest.param(
+            True,
+            "redirect",
+            "any(str(x).endswith('.so') for x in files(pkg.subpkg2.subsubpkg1).iterdir())",
+            marks=pytest.mark.xfail,
+        ),
+        (True, "redirect", "isinstance(files(pkg.subpkg2.subsubpkg2), pathlib.Path)"),
+        pytest.param(
+            True,
+            "redirect",
+            "any(str(x).endswith('.so') for x in files(pkg.subpkg2.subsubpkg2).iterdir())",
+            marks=pytest.mark.xfail,
+        ),
+        # Editable inplace
+        (True, "inplace", "isinstance(files(pkg), pathlib.Path)"),
+        (True, "inplace", "any(str(x).endswith('.so') for x in files(pkg).iterdir())"),
+        (True, "inplace", "isinstance(files(pkg.subpkg1), pathlib.Path)"),
+        (
+            True,
+            "inplace",
+            "any(str(x).endswith('.so') for x in files(pkg.subpkg1).iterdir())",
+        ),
+        (True, "inplace", "isinstance(files(pkg.subpkg2), pathlib.Path)"),
+        (
+            True,
+            "inplace",
+            "any(str(x).endswith('.so') for x in files(pkg.subpkg2).iterdir())",
+        ),
+        (True, "inplace", "isinstance(files(pkg.subpkg2.subsubpkg1), pathlib.Path)"),
+        (
+            True,
+            "inplace",
+            "any(str(x).endswith('.so') for x in files(pkg.subpkg2.subsubpkg1).iterdir())",
+        ),
+        (True, "inplace", "isinstance(files(pkg.subpkg2.subsubpkg2), pathlib.Path)"),
+        (
+            True,
+            "inplace",
+            "any(str(x).endswith('.so') for x in files(pkg.subpkg2.subsubpkg2).iterdir())",
+        ),
+    ],
+)
+def test_importlib_resources(
+    monkeypatch, tmp_path, editable, editable_mode, isolated, check
+):
+    _setup_package_for_editable_layout_tests(  # type: ignore[no-untyped-call]
+        monkeypatch, tmp_path, editable, editable_mode, isolated
+    )
+    value = isolated.execute(
+        textwrap.dedent(
+            f"""
+            from importlib.resources import files
+            from importlib.readers import MultiplexedPath
+            import pkg
+            import pathlib
+            print({check})
```

Comment:
This should be `assert` instead of `print`. Could you also move `check` to be in-lined here instead of a parameter?

Reply:
Rather than asserting here, I print the boolean result and assert the string value outside because asserting inside a subprocess requires significantly more post-processing to handle cleanly.

Reply:
Could you elaborate on why it requires more post-processing, or what the output you get when doing that?

Reply:
I take it back. I originally said this because while debugging specific cases it was more challenging to extract information from subprocess exceptions than to simply parse inputs and outputs to run inside the environment, but in the final state the output is sufficiently readable as is. I've moved the assert.

> Could you also move check to be in-lined here instead of a parameter?

I don't understand this suggestion. `check` is parametrized, so how would I inline it without duplicating the test?

Reply:
What I was suggesting is to not parametrize it, and instead write all of the checks explicitly in the `isolated.execute` part

Reply:
I guess we could do that. Would you just want each of the parametrizations to be a separate test, then, with all the parameters hardcoded within each test (and presumably in the name)?

Reply:
Not like that. Just put the asserts one after another and let it run until it fails the first time.

There is `pytest_check` if you want to get a report for multiple commands, but I don't think we will gain much by using that

Reply:
Oh OK yes I can do that, sure.

When you get a chance, let me know what you think of the latest changes and https://github.com/scikit-build/scikit-build-core/pull/906#issuecomment-2423365100, then I can push the next update to this PR and address this request as well.

Reply:
Yes, busy week, but from Monday I will have time